test/utils/server.js

```diff
@@ -305,9 +305,9 @@ export default class TestServer {
 		if (p === '/error/premature') {
 			res.writeHead(200, {'content-length': 50});
 			res.write('foo');
-			setTimeout(() => {
-				res.destroy();
-			}, 100);
+			setImmediate(() => {
```

Comment:
Hmm, haven't seen setImmediate being used for a long time anywhere in the wild. Thought it was more a IE thing. after all it seems to be "[none standard](https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate)"

guess what most node developer use is `process.nextTick()` but this is the test file so it don't even mather so much.